core/src/main/scala/com/mesosphere/usi/core/CachedPendingLaunch.scala

```diff
@@ -0,0 +1,26 @@
+package com.mesosphere.usi.core
+import com.mesosphere.usi.core.models.PodId
+
+case class CachedPendingLaunch(pendingLaunch: Set[PodId] = Set.empty) {
```

Comment:
This looks like premature optimization to me. Is there a benchmark showing that filtering a list of pod specs actually takes long? How many pod specs do you need to see the difference and how big is it? Is it on the order of μs? ms?

Reply:
Discussed; well aware of premature optimization arguments. It is obvious that we will need caching mechanisms, and I wanted to work out how they will look early in the development of the SchedulerLogic. Recomputing pending launch states for every message is nonsense. I'd like to keep it.